From 0c8977b1869311cde56fd0494cdeacf5feb3a603 Mon Sep 17 00:00:00 2001
Message-Id: <0c8977b1869311cde56fd0494cdeacf5feb3a603.1423724298.git.chang-joon.lee@intel.com>
From: Uma Shankar <uma.shankar@intel.com>
Date: Mon, 5 Jan 2015 17:34:28 +0530
Subject: [PATCH] FOR_UPSTREAM [VPG]: drm/i915: Resolved GPIO Sideband
 programming issue

To acces GPIO and GPS sideband registers, devfn as device 0, function 0
i.e, DEVFN(0, 0) should be used. These were incorrectly been accessed
with device 2 function 0. Rectified the same.

Issue: IRDA-2793
Change-Id: I010b10b93b1525c2704d416f41da376cb34054e2
Signed-off-by: Uma Shankar <uma.shankar@intel.com>
Signed-off-by: Shen, Lei <lei.shen@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h       |    2 ++
 drivers/gpu/drm/i915/intel_sideband.c |    8 ++++----
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index edad7a2..e6ea3bf 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -802,6 +802,8 @@ enum punit_power_well {
  * -----------------
  */
 #define DPIO_DEVFN			0
+#define GPIO_DEVFN			0
+#define GPS_DEVFN			0
 
 #define DPIO_CTL			(VLV_DISPLAY_BASE + 0x2110)
 #define  DPIO_MODSEL1			(1<<3) /* if ref clk b == 27 */
diff --git a/drivers/gpu/drm/i915/intel_sideband.c b/drivers/gpu/drm/i915/intel_sideband.c
index 9d79f16..6a3d4d0 100644
--- a/drivers/gpu/drm/i915/intel_sideband.c
+++ b/drivers/gpu/drm/i915/intel_sideband.c
@@ -137,7 +137,7 @@ u32 vlv_nc_read(struct drm_i915_private *dev_priv, u8 addr)
 u32 vlv_gpio_read(struct drm_i915_private *dev_priv, u8 core_offset, u32 reg)
 {
 	u32 val = 0;
-	vlv_sideband_rw(dev_priv, PCI_DEVFN(2, 0), core_offset,
+	vlv_sideband_rw(dev_priv, GPIO_DEVFN, core_offset,
 			SB_CRRDDA_NP, reg, &val);
 	return val;
 }
@@ -145,7 +145,7 @@ u32 vlv_gpio_read(struct drm_i915_private *dev_priv, u8 core_offset, u32 reg)
 void vlv_gpio_write(struct drm_i915_private *dev_priv, u8 core_offset,
 				u32 reg, u32 val)
 {
-	vlv_sideband_rw(dev_priv, PCI_DEVFN(2, 0), core_offset,
+	vlv_sideband_rw(dev_priv, GPIO_DEVFN, core_offset,
 			SB_CRWRDA_NP, reg, &val);
 }
 
@@ -180,14 +180,14 @@ void vlv_ccu_write(struct drm_i915_private *dev_priv, u32 reg, u32 val)
 u32 vlv_gps_core_read(struct drm_i915_private *dev_priv, u32 reg)
 {
 	u32 val = 0;
-	vlv_sideband_rw(dev_priv, PCI_DEVFN(2, 0), IOSF_PORT_GPS_CORE,
+	vlv_sideband_rw(dev_priv, GPS_DEVFN, IOSF_PORT_GPS_CORE,
 			SB_CRRDDA_NP, reg, &val);
 	return val;
 }
 
 void vlv_gps_core_write(struct drm_i915_private *dev_priv, u32 reg, u32 val)
 {
-	vlv_sideband_rw(dev_priv, PCI_DEVFN(2, 0), IOSF_PORT_GPS_CORE,
+	vlv_sideband_rw(dev_priv, GPS_DEVFN, IOSF_PORT_GPS_CORE,
 			SB_CRWRDA_NP, reg, &val);
 }
 
-- 
1.7.9.5

